src/components/structures/MessagePanel.js

```diff
@@ -388,6 +388,8 @@ module.exports = React.createClass({
                 isVisibleReadMarker = visible;
             }
 
+            // XXX: there should be no need for a ghost tile - we should just use a
```

Comment:
From what I can see, the ghost tile is the one that animates and the `ref` on the ghost is used to start the animation. This could be so much better.